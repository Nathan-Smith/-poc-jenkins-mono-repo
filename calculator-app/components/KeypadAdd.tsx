import { KeypadButton } from './KeypadButton'

export function KeypadAdd(): JSX.Element {
  return (
    <KeypadButton testid="keypad-add" gridArea="keypad-add">
      +
    </KeypadButton>
  )
}
