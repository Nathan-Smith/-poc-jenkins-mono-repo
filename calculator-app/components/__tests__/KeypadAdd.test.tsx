import React from 'react'
import renderer, { act, ReactTestRenderer } from 'react-test-renderer'

import { KeypadAdd } from '../KeypadAdd'

describe('KeypadAdd', () => {
  test('renders', async () => {
    let component: ReactTestRenderer | undefined
    act(() => {
      component = renderer.create(<KeypadAdd/>)
    })

    expect(component?.toJSON()).toMatchSnapshot()
  })
})
