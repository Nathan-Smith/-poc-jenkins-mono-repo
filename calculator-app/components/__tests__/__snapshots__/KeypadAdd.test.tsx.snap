// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeypadAdd renders 1`] = `
<button
  className="mdc-button mdc-button--outlined keypadButton"
  data-testid="keypad-add"
  onClick={[Function]}
  style={
    Object {
      "gridArea": "keypad-add",
    }
  }
>
  <span
    className="mdc-button__ripple"
  />
  <span
    className="mdc-button__label"
  >
    +
  </span>
</button>
`;
